internal/provider/resource_gitlab_project.go

```diff
@@ -1259,12 +1259,7 @@ func resourceGitlabProjectRead(ctx context.Context, d *schema.ResourceData, meta
 func resourceGitlabProjectUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*gitlab.Client)
 
-	// Always send the name field, to satisfy the requirement of having one
-	// of the project attributes listed below in the update call
-	// https://gitlab.com/gitlab-org/gitlab-foss/-/blob/master/lib/api/helpers/projects_helpers.rb#L120-188
-	options := &gitlab.EditProjectOptions{
-		Name: gitlab.String(d.Get("name").(string)),
-	}
+	options := &gitlab.EditProjectOptions{}
```

Comment:
Just "removing" it would probably lead to this bug again #1157 . WDYT? And it would also disallow for owners to change the name.

Reply:
You're right #1157 might happen again with this. But I don't see how owners would be disallowed to change the name, the field is still sent in line 1266. 

According to [this](https://gitlab.com/gitlab-org/gitlab-foss/-/blob/master/lib/api/helpers/projects_helpers.rb#L120-188), the fields that have to be passed with at least the `name` field are the following:
```
['only_allow_merge_if_all_status_checks_passed', 'approvals_before_merge', 'external_authorization_classification_label', 'import_url', 'issues_template', 'mirror_overwrites_diverged_branches', 'mirror_trigger_builds', 'mirror_user_id', 'mirror', 'only_mirror_protected_branches', 'requirements_access_level', 'wiki_enable']
```

What if we check if the request only has these fields, and in that case we add the name field explicitely?

Reply:
I've pushed a new commit with this change. What do you think? @timofurrer